src/marqo/tensor_search/tensor_search.py

```diff
@@ -627,10 +635,30 @@ def add_documents(config: Config, index_name: str, docs: List[dict], auto_refres
                     })
 
             # Add chunks_to_append along with doc metadata to total chunks
+
+            elif isinstance(field_content, dict):
+
+                combo_chunk, combo_document_is_valid, unsuccessful_doc_to_append, combo_vectorise_time_to_add = \
+                    vectorise_multimodal_combination_field(field, field_content, copied,
+                                                           i, doc_id, selected_device, index_info, image_repo)
+
+                multimodal_field_for_filter[field] = list(field_content)
```

Comment:
Can we just directly add it to the copied doc?

Reply:
no we can't because we are iterating through copied here.